source/slang/syntax.cpp

```diff
@@ -2061,4 +1918,53 @@ void Type::accept(IValVisitor* visitor, void* extra)
         }*/
         return newSubst;
     }
+    bool SubstitutionSet::Equals(SubstitutionSet substSet) const
+    {
+        if (genericSubstitutions)
+        {
+            if (!genericSubstitutions->Equals(substSet.genericSubstitutions))
+                return false;
+        }
+        else
+        {
+            if (substSet.genericSubstitutions)
+                return false;
+        }
+        if (thisTypeSubstitution)
+        {
+            if (!thisTypeSubstitution->Equals(substSet.thisTypeSubstitution))
+                return false;
+        }
+        else
+        {
+            if (substSet.thisTypeSubstitution && substSet.thisTypeSubstitution->sourceType != nullptr)
+                return false;
+        }
+        return true;
```

Comment:
Why don't we compare global generic substitutions here? Is it just that they should never matter, or is it that things would actually go wrong?